Implement Rayleigh RV as a scaled Chi2

### Description

Code:
```python
def rayleigh(scale=1.0, *, size=None, **kwargs):
  return chi2(df=2, size=size, **kwargs) * as_tensor_variable(scale)
```

from:
https://github.com/pymc-devs/pytensor/pull/220#issuecomment-1435947464

Using a code along the above mentioned lines:
```
def rayleigh(scale=1.0, *, size=None, **kwargs):
  return chisquare(df=2, size=size, **kwargs) * as_tensor_variable(scale)
```
, the test:
```
def test_rayleigh_samples():
    compare_sample_values(rayleigh)
```
fails with traceback:
```
____________________________ test_rayleigh_samples ____________________________

    def test_rayleigh_samples():
>       compare_sample_values(rayleigh)

tests\tensor\random\test_basic.py:401:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\tensor\random\test_basic.py:138: in compare_sample_values
    np.testing.assert_allclose(pytensor_res_val, numpy_res)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<function assert_allclose.<locals>.compare at 0x00000187EBDF4180>, array(1.58225191), array(1.25787595))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-07, atol=0', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=0
E
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 0.32437596
E           Max relative difference: 0.25787595
E            x: array(1.582252)
E            y: array(1.257876)```
```
A test more along the chi2 lines also fails like this:
```
@pytest.mark.parametrize(
    "size",
    [
        (None),
        ([]),
    ],
)
def test_rayleigh_samples(size):
    compare_sample_values(rayleigh, size=size, test_fn=fixed_scipy_rvs("rayleigh"))
```
with traceback:
```
_________________________ test_rayleigh_samples[None] _________________________

size = None

    @pytest.mark.parametrize(
        "size",
        [
            (None),
            ([]),
        ],
    )
    def test_rayleigh_samples(size):
>       compare_sample_values(rayleigh, size=size, test_fn=fixed_scipy_rvs("rayleigh"))

tests\tensor\random\test_basic.py:401:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\tensor\random\test_basic.py:138: in compare_sample_values
    np.testing.assert_allclose(pytensor_res_val, numpy_res)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<function assert_allclose.<locals>.compare at 0x0000011B227AA020>, array(0.47892931), array(0.69204719))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-07, atol=0', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=0
E
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 0.21311788
E           Max relative difference: 0.30795281
E            x: array(0.478929)
E            y: array(0.692047)

..\..\..\miniconda3\envs\pytensor-dev\Lib\contextlib.py:81: AssertionError
________________________ test_rayleigh_samples[size1] _________________________

size = []

    @pytest.mark.parametrize(
        "size",
        [
            (None),
            ([]),
        ],
    )
    def test_rayleigh_samples(size):
>       compare_sample_values(rayleigh, size=size, test_fn=fixed_scipy_rvs("rayleigh"))

tests\tensor\random\test_basic.py:401:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests\tensor\random\test_basic.py:138: in compare_sample_values
    np.testing.assert_allclose(pytensor_res_val, numpy_res)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<function assert_allclose.<locals>.compare at 0x0000011B228BB880>, array(1.7294089), array(1.31506992))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-07, atol=0', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=0
E
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 0.41433898
E           Max relative difference: 0.31506992
E            x: array(1.729409)
E            y: array(1.31507)

```

--- a/pytensor/tensor/random/basic.py
+++ b/pytensor/tensor/random/basic.py
@@ -6,6 +6,7 @@ import scipy.stats as stats
 
 import pytensor
 from pytensor.tensor.basic import arange, as_tensor_variable
+from pytensor.tensor.math import sqrt
 from pytensor.tensor.random.op import RandomVariable
 from pytensor.tensor.random.type import RandomGeneratorType, RandomStateType
 from pytensor.tensor.random.utils import (
@@ -524,6 +525,43 @@ def chisquare(df, size=None, **kwargs):
         is returned.
     """
     return gamma(shape=df / 2.0, scale=2.0, size=size, **kwargs)
+
+
+def rayleigh(scale=1.0, *, size=None, **kwargs):
+    r"""Draw samples from a Rayleigh distribution.
+
+    The probability density function for `rayleigh` with parameter `scale` is given by:
+
+    .. math::
+        f(x; s) = \frac{x}{s^2} e^{-x^2/(2 s^2)}
+
+    where :math:`s` is the scale parameter.
+
+    This variable is obtained by taking the square root of the sum of the squares of
+    two independent, standard normally distributed random variables.
+
+    Signature
+    ---------
+    `() -> ()`
+
+    Parameters
+    ----------
+    scale : float or array_like of floats, optional
+        Scale parameter of the distribution (positive). Default is 1.0.
+    size : int or tuple of ints, optional
+        Output shape. If the given shape is, e.g., `(m, n, k)`, then `m * n * k` samples
+        are drawn. Default is None, in which case the output shape is determined by the
+        shape of `scale`.
+
+    Notes
+    -----
+    `Rayleigh` is a special case of `chisquare` with ``df=2``.
+    """
+
+    scale = as_tensor_variable(scale)
+    if size is None:
+        size = scale.shape
+    return sqrt(chisquare(df=2, size=size, **kwargs)) * scale
 
 
 class ParetoRV(ScipyRandomVariable):

--- a/tests/tensor/random/test_basic.py
+++ b/tests/tensor/random/test_basic.py
@@ -50,6 +50,7 @@ from pytensor.tensor.random.basic import (
     permutation,
     poisson,
     randint,
+    rayleigh,
     standard_normal,
     t,
     triangular,
@@ -388,6 +389,20 @@ def test_gamma_deprecation_wrapper_fn():
 )
 def test_chisquare_samples(df, size):
     compare_sample_values(chisquare, df, size=size, test_fn=fixed_scipy_rvs("chi2"))
+
+
+@pytest.mark.parametrize(
+    "scale, size",
+    [
+        (1, None),
+        (2, []),
+        (4, 100),
+    ],
+)
+def test_rayleigh_samples(scale, size):
+    compare_sample_values(
+        rayleigh, scale=scale, size=size, test_fn=fixed_scipy_rvs("rayleigh")
+    )
 
 
 @pytest.mark.parametrize(

